lab_2/lab_work_2.md

```diff
@@ -0,0 +1,195 @@
+> Подсказка №2: внимательно ознакомьтесь с Шагом 4, чтобы оценить требуемый размер матрицы.
+
+Например матрица из 3 строк и 4 столбцов:
+
+```py
+[
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0]
+]
+```
+
+Интерфейс: 
+
+```py
+def generate_edit_matrix(num_rows: int, num_cols: int) -> list:
+  pass
+```
+
+### Шаг 2. Инициализация матрицы решений
+
+Расчёт минимального расстояния выполняется только по предварительно инициализированной
+матрице решений. Инициализация заключается в заполнении матрицы значениями,
+соответствующими работе с пустыми строками.
+
+Функция принимает матрицу в качестве единственного аргумента и возвращает
+измененную версию этой матрицы.
+
+Правила инициализации матрицы:
+
+* заполнение первого стобца матрицы
+<img src="https://latex.codecogs.com/gif.latex?\begin{cases}&space;\right.M[0][0]&space;=&space;0&space;\\&space;M[i][0]&space;=&space;M[i-1][0]&space;&plus;&space;remove\_weight&space;\end{cases}" title="\begin{cases} \right.M[0][0] = 0 \\ M[i][0] = M[i-1][0] + remove\_weight \end{cases}" />
+* заполнение первой строки матрицы
+<img src="https://latex.codecogs.com/gif.latex?\begin{cases}&space;\right.M[0][0]&space;=&space;0&space;\\&space;M[0][j]&space;=&space;M[0][j-1]&space;&plus;&space;insert\_weight&space;\end{cases}" title="\begin{cases} \right.M[0][0] = 0 \\ M[0][j] = M[0][j-1] + insert\_weight \end{cases}" />
+
+Интерфейс:
+
+```py
+def initialize_edit_matrix(edit_matrix: tuple, add_weight: int, remove_weight: int) -> list:
+  pass
+```
+
+### Шаг 3. Найти наименьшее значение из набора заданных чисел
+
+Функция принимает на вход кортеж из чисел и возвращает наименьшее из них.
+
+Интерфейс:
+
+```py
+def minimum_value(numbers: tuple) -> int:
+  pass
+```
+
+### Шаг 4. Заполнить матрицу решений
+
+Функция принимает на вход инициализированную матрицу решений и веса значимости операций.
+
+Правило заполнения каждой ячейки матрицы:
+<img src="https://latex.codecogs.com/gif.latex?M[i][j]&space;=&space;min\begin{cases}&space;M[i-1][j]&space;&plus;&space;remove\_weight&space;\\&space;M[i][j-1]&space;&plus;&space;insert\_weight&space;\\&space;M[i-1][j-1]&space;&plus;&space;\begin{cases}&space;substitute\_weight&space;\text{,&space;if&space;original\_word[i]}&space;\neq&space;\text{target\_word[j]}&space;\\&space;0&space;\text{,&space;if&space;original\_word[i]}&space;=&space;\text{target\_word[i]}&space;\\&space;\end{cases}&space;\end{cases}" title="M[i][j] = min\begin{cases} M[i-1][j] + remove\_weight \\ M[i][j-1] + insert\_weight \\ M[i-1][j-1] + \begin{cases} substitute\_weight \text{, if original\_word[i]} \neq \text{target\_word[j]} \\ 0 \text{, if original\_word[i]} = \text{target\_word[i]} \\ \end{cases} \end{cases}" />
+
+> Подсказка: при итерировании по матрице обратите внимание на диапазоны счётчиков
+> (переменные цикла) - какое они могут принимать минимальное и макимальное
+> значения
+
+Интерфейс:
+
+```py
+def fill_edit_matrix(edit_matrix: tuple,
```

Comment:
The same about `edit_matrix`

Reply:
I think we also need to pass words here as we need to check the similarity of letters for substitution